packages/commerce-sdk-react/src/components/StorefrontPreview/storefront-preview.tsx

```diff
@@ -51,6 +53,25 @@ export const StorefrontPreview = ({
         }
     }, [enabled, getToken, onContextChange])
 
+    useEffect(() => {
+        if (enabled && isHostTrusted) {
+            // In Storefront Preview mode, add cache breaker for the SCAPI's GET requests.
+            // Otherwise, it's possible to get stale responses after the Shopper Context is set.
+            // (i.e. in this case, we optimize for accurate data, rather than performance/caching)
+            proxyGetRequests(apiClients, {
+                apply(target, thisArg, argumentsList) {
+                    if (argumentsList.length === 0) {
+                        argumentsList[0] = {parameters: {c_cache_breaker: new Date().getTime()}}
+                    } else {
+                        argumentsList[0].parameters = argumentsList[0].parameters ?? {}
+                        argumentsList[0].parameters.c_cache_breaker = new Date().getTime()
+                    }
```

Comment:
You can access and assign an array of 0 length without error. E.g. var a = []; a[10] = 'happy birthday Vincent' will result in 10 empty entries and a 10th with that messages.

So you can probably simplify this logic knowing this too look something like this (not tested)

```suggestion
                    argumentsList[0] = {
                           ...argumentsList[0],
                           parameters: {
                               ...argumentsList[0].parameters,
                               c_cache_breaker: new Date().getTime()
                           }
                    }
```

Reply:
Thanks Ben, yup that's clearer code. Just need to modify it a bit by considering that `parameters` may not exist:
```js
parameters: {
    ...argumentsList[0]?.parameters,
```